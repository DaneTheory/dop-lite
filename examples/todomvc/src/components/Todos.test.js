import React from 'react'
import renderer from 'react-test-renderer';
import Todos from './Todos'
import { addTodo } from '../actions'

let component
test('Adding a new todo', () => {
  component = renderer.create(
    <Todos />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // Adding todo manually
  const labelnewtodo = 'My new todo'
  addTodo(labelnewtodo)
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // Checking the todo has been added
  let todoelement = tree.children[0].children[0].children[0].children[0].children[1]
  expect(todoelement.type).toMatchSnapshot('label');
  expect(todoelement.children[0]).toMatchSnapshot(labelnewtodo);
});


test('Removing todo', () => {
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  
  // removeTodo manually
  let buttonchildrencomponent = tree.children[0].children[0].children[0].children[0].children[2]
  buttonchildrencomponent.props.onClick()

  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree.children[0].children[0].children).toBe(null);
})
