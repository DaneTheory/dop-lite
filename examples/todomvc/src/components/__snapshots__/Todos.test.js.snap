// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Adding a new todo 1`] = `
<div>
  <section
    className="main"
  >
    <ul
      className="todo-list"
    />
  </section>
</div>
`;

exports[`Adding a new todo 2`] = `
<div>
  <section
    className="main"
  >
    <ul
      className="todo-list"
    >
      <li
        className=""
      >
        <div
          className="view"
        >
          <input
            checked={false}
            className="toggle"
            onChange={[Function]}
            type="checkbox"
          />
          <label
            onDoubleClick={[Function]}
          >
            My new todo
          </label>
          <button
            className="destroy"
            onClick={[Function]}
          />
        </div>
      </li>
    </ul>
  </section>
</div>
`;

exports[`My new todo 1`] = `"My new todo"`;

exports[`Removing todo 1`] = `
<div>
  <section
    className="main"
  >
    <ul
      className="todo-list"
    >
      <li
        className=""
      >
        <div
          className="view"
        >
          <input
            checked={false}
            className="toggle"
            onChange={[Function]}
            type="checkbox"
          />
          <label
            onDoubleClick={[Function]}
          >
            My new todo
          </label>
          <button
            className="destroy"
            onClick={[Function]}
          />
        </div>
      </li>
    </ul>
  </section>
</div>
`;

exports[`Removing todo 2`] = `
<div>
  <section
    className="main"
  >
    <ul
      className="todo-list"
    />
  </section>
</div>
`;

exports[`label 1`] = `"label"`;
